series/type-time.go

```diff
@@ -0,0 +1,150 @@
+package series
+
+import (
+	"fmt"
+	"time"
+)
+
+// timeElement is the concrete implementation of the Element interface for
+// time.Time. If the stored time.Time is zero, it will be considered as a NaN
+// element.
+type timeElement struct {
+	e   time.Time
+	nan bool
+}
+
+func (e *timeElement) Set(value interface{}) {
+	e.nan = false
+	switch value.(type) {
+	case string:
+		v := string(value.(string))
+		var err error
+		e.e, err = time.Parse(timeformat, v)
+		if err != nil {
+			e.e = time.Time{}
+		}
+		return
+	case int, float64, bool:
+		e.e = time.Time{}
+	case time.Time:
+		e.e = value.(time.Time)
+	case Element:
+		e.e, _ = value.(Element).Time()
+	default:
+		e.nan = true
+		return
+	}
+	return
+}
+
+func (e timeElement) Copy() Element {
+	if e.nan {
+		return &timeElement{time.Time{}, true}
+	}
+	return &timeElement{e.e, false}
+}
+
+func (e timeElement) IsNA() bool {
+	return e.e.IsZero()
+}
+
+func (e timeElement) Type() Type {
+	return Time
+}
+
+func (e timeElement) Val() ElementValue {
+	if e.IsNA() {
+		return nil
+	}
+	return e.e
+}
+
+func (e timeElement) String() string {
+	if e.nan {
+		return "NaN"
+	}
+	return e.e.Format(timeformat)
+}
+
+func (e timeElement) Int() (int, error) {
+	if e.IsNA() {
+		return 0, fmt.Errorf("timeElement.Int(): can't convert NaN to int")
+	}
+	return int(e.e.Unix()), nil
```

Comment:
What happens here if e.e.Unix overflows int? (int is 32b instead of 64b) Need to check this before making the conversion and throw an error if necessary.

Also, are we sure we want `Unix` (Seconds from Epoch) or do we want `UnixNano` (Nanoseconds from Epoch)? Maybe we could discuss this on the data-science slack channel...